frontend/src/components/ViewActivities/activityfns.js

```diff
@@ -54,4 +57,30 @@ export function getField(activity, fieldName, defaultValue){
     return defaultValue;
   }
   return activity[fieldName];
+}
+
+/**
+ * Write contents into an activity in the database.
+ * 
+ * @param {string} tripId Database ID of the trip whose actiivty should be modified.
+ * @param {string} activityId Database ID of the activity to be modified.
+ * @param {Object} newValues Dictionary of the new values in {fieldName: newValue} form
+ */
+export async function writeActivity(tripId, activityId, newValues) {
+  // todo: check if tripId or activityId is not valid. (#58)
+  if (newValues === null) {
```

Comment:
Is this ever possible? I pretty sure that at a minimum it will just be an object with at least one key:val pair given your instantiation of `newVals` and the call to `writeActivity` in `activity.js`.

Reply:
Yeah, if the user just looks at a trip and then submits the form without changing anything. I'll put a "timestamp" field in there just so it's never empty.

Reply:
nope. If the user submits an empty form .current.value is "". Fixed.